readmeio/rdme
# feat: add `categories` and `categories:create`

| 🚥 Fix RM-XXX |
| :-- |

## 🧰 Changes

Added support for two new commands `categories` and `categories:create`. 
- `categories` returns a raw json of all categories for a version including pagination.
- `categories:create` allows you to create a new category with a specified `title` and `type` for your project version. The `title` will be converted to a slug in param case. If a category already exists with the same slug and type for
the specified version then a new category is not created and the id of the existing category is returned. This includes if the slug contains incremented suffix such as `-1` or `-2`. If there are multiple categories with a matching slug and type then only one matching category is returned.

## 🧬 QA & Testing

- `categories`
    1. Login to a ReadMe project with `rdme login`
    2. Run `rdme categories`
- `categories:create`
   1. Login to readme project with `rdme login`
   2. Run `rdme categories:create "test title" --categoryType guide` and see a new category was created with a slug of `test-title`
   3. Run `rdme categories:create "test title" --categoryType guide` and see a new category was not created but rather it returns the slug and id of the category created in the previous step.
   4. Run `rdme categories:create "test title" --categoryType reference` and see a new category was created with a slug of `test-title-1`
   5. Run `rdme categories:create "test title" --categoryType reference` and see a new category was not created but rather it returns the slug and id of the category created in the previous step.
   6.  Run `rdme categories:create "test title"` and see a `categoryType` of "guide" or "reference" is required
